src/sage/rings/species.py

```diff
@@ -0,0 +1,2337 @@
+        """
+        if self.parent()._arity == 1:
+            return "{" + f"{self._dis.gens()}" + "}"
+        dompart = ', '.join("{" + repr(sorted(b))[1:-1] + "}"
+                            for b in self._dompart)
+        return "{" + f"{self._dis.gens()}: ({dompart})" + "}"
+
+    def grade(self):
+        r"""
+        Return the grade of ``self``.
+
+        EXAMPLES::
+
+            sage: from sage.rings.species import AtomicSpecies
+            sage: A = AtomicSpecies("X, Y")
+            sage: G = PermutationGroup([[(1,2),(3,4),(5,6),(7,8,9,10)]])
+            sage: a = A(G, {0: [1,2,3,4], 1: [5,6,7,8,9,10]})
+            sage: a.grade()
+            [4, 6]
+        """
+        S = self.parent().grading_set()
+        return S(self._mc)
+
+    def __lt__(self, other):
+        r"""
+        Return whether ``self`` is less than ``other``.
+
+        ``self`` is less than or equal to ``other`` if it is
+        conjugate to a subgroup of ``other`` in the parent group.
+
+        EXAMPLES::
+
+            sage: from sage.rings.species import AtomicSpecies
+            sage: A = AtomicSpecies("X")
+            sage: A(DihedralGroup(4)) < A(CyclicPermutationGroup(4))
+            True
+
+        We create the poset of atomic species of degree four::
+
+            sage: P = Poset([A.subset(4), lambda b, c: b <= c])
+            sage: len(P.cover_relations())
+            7
+            sage: P.cover_relations()  # random
+            [[E_4, Eo_4],
+             [E_4, P_4],
+             [Eo_4, Pb_4],
+             [P_4, C_4],
+             [P_4, Pb_4],
+             [C_4, {((1,2)(3,4),)}],
+             [Pb_4, {((1,2)(3,4),)}]]
+
+        TESTS::
+
+            sage: A = AtomicSpecies("X")
+            sage: [(a, b) for a, b in Subsets(A.subset(4), 2) if (a < b) != (b > a)]
+            []
+            sage: [(a, b) for a, b in Subsets(A.subset(4), 2) if (a <= b) != (b >= a)]
+            []
+            sage: A = AtomicSpecies("X, Y")
+            sage: [(a, b) for a, b in Subsets(A.subset(3), 2) if (a < b) != (b > a)]
+            []
+        """
+        # the arities match because the parents are equal
+        if self._mc != other._mc:
+            # X should come before Y
+            return (sum(self._mc) < sum(other._mc)
+                    or (sum(self._mc) == sum(other._mc)
+                        and self._mc > other._mc))
+        S = SymmetricGroup(sum(self._mc)).young_subgroup(self._mc)
+        # conjugate self and other to match S
+        g = list(chain.from_iterable(self._dompart))
+        conj_self = PermutationGroupElement(g).inverse()
+        G = libgap.ConjugateGroup(self._dis, conj_self)
+        h = list(chain.from_iterable(other._dompart))
+        conj_other = PermutationGroupElement(h).inverse()
+        H = libgap.ConjugateGroup(other._dis, conj_other)
+        return GAP_FAIL != libgap.ContainedConjugates(S, G, H, True)
+
+    def __le__(self, other):
```

Comment:
It's fine to have this in here, but I thought it wasn't necessary to implement it.